### DISCUSSION TITLE
如果创建dom时，数据为空，可不可以不报错

### DISCUSSION THREAD

**Discussion by cthulhu-tww** [vanjs-org/van#54] [Ideas]
首先感谢香草js帮助我的谷歌浏览器扩展程序节约了一半的代码。但是在使用过程中，我发现有个很令人费解的问题，我不知道是有意设计成这样还是如何。当我bind 一个object或者其它数据类型且当数据中出现了 null 或者 undefined时，我的js就会报错，导致我整个dom就会崩溃。这就会导致我必须不停的给object本身及其字段加上非空判断，并且加上“” 这种空字符串来保证我的js 不发生报错。而我的想法是如果数据为空，不如就仅仅是这个dom及其子dom不会渲染，而不必使整个dom树损坏不是吗。

**Comment by Tao-VanJS** [comment]
嗯，目前这个是intended behavior。因为tag functions不支持`null`或`undefined`作为child node。目前有一个简单的workaround：`dom ?? []`。在后续版本中可能会考虑加上对`null`或者`undefined`作为child node的支持。

**Comment by Tao-VanJS** [comment]
FYI, `null` and `undefined` child node is supported in [tag functions](https://vanjs.org/tutorial#api-tags) and [`van.add`](https://vanjs.org/tutorial#api-add) in [0.12.1 release](https://github.com/vanjs-org/van/discussions/53#discussioncomment-6137688).

**Comment by cthulhu-tww** [comment]
感谢，我非常期待新的版本

**Reply by Tao-VanJS** [reply]
They're already available now:-) See the release notes: https://github.com/vanjs-org/van/discussions/53#discussioncomment-6137688

**Reply by cthulhu-tww** [reply]
哦哦，我这就去调整我的代码，感谢回复！
